src/extension.ts

```diff
@@ -36,13 +45,49 @@ function searchSpecific() {
 function searchAll() {
   var editor = getEditor();
   var query = getSelectedText(editor);
-
   var dash = new Dash(OS);
 
   exec(dash.getCommand(query));
 }
 
 /**
+ * Search in dash for editor syntax documentation
+ */
+function searchEmptySyntax() {
+
+  var editor = getEditor();
+  var query = "";
+  var languageId = editor.document.languageId;
+  var docsets = getDocsets(languageId);
+  var dash = new Dash(OS);
+
+  exec(dash.getCommand(query, docsets));
+}
+
+/**
+ * Search in dash for editor syntax documentation with a custom query
+ */
+function searchCustomWithSyntax() {
+
+  var editor = getEditor();
+  var languageId = editor.document.languageId;
+  var docsets = getDocsets(languageId);
+  var dash = new Dash(OS);
+
+  var inputOptions: InputBoxOptions = {
+    placeHolder: "Something to search in Dash.",
+    prompt: "Enter something to search for in Dash."
+  }
+
+  window.showInputBox(inputOptions)
+    .then((query: string) => {
+      console.log(query);
+      if (query) //If they actually input code
+        exec(dash.getCommand(query, docsets)); //Open it in dash

```

Comment:
don't forget the curly braces. I think I should set up a lint in future to prevent this.